Generals/Code/GameEngine/Include/GameNetwork/NetworkDefs.h

```diff
@@ -170,7 +170,12 @@ enum PlayerLeaveCode CPP_11(: Int) {
 };
 
 // Magic number for identifying a Generals packet.
+#if defined(_MSC_VER) && _MSC_VER < 1300
+// Original retail value is used for builds with VC6
 static const UnsignedShort GENERALS_MAGIC_NUMBER = 0xF00D;
+#else
+static const UnsignedShort GENERALS_MAGIC_NUMBER = 0xF00D+1;
```

Comment:
Thinking about this again, this means new clients will not see retail client, right? That means some players cannot see a group of other players, which may or may not be desirable.

Ideally, all clients would be able to see all other client rooms in the lobby, but not be able to join them if incompatible. There are perhaps tricks we can do to prevent retail clients messing with rooms of new clients. Preventing new clients from joining retail client rooms should be possible.

Reply:
From user experience it means 2 players can go into the same network and not see each other, then potentially struggle to find out what the reason for it is. If they would be able to see each other and our client would tell the user that he cannot connect with the other guy because he is using an incompatible client, then this would be a better user experience.